### PR TITLE
more code grooming

### PR DESCRIPTION
**Pull request by orklah**
It's pretty minor refactoring, except for what I removed in ParseTreeCreator. What I removed felt important but I couldn't get how it could be useful and tests seems to prove me right...

### CODE REVIEW

File: src/Psalm/Internal/Provider/ReturnTypeProvider/FilterVarReturnTypeProvider.php [hunk 1]
```diff
@@ -55,8 +55,6 @@ public static function getFunctionReturnType(FunctionReturnTypeProviderEvent $ev
         ) {
             $filter_type_type = $second_arg_type->getSingleIntLiteral();
 
-            $filter_type = null;
```
Comment (weirdan) [inline src/Psalm/Internal/Provider/ReturnTypeProvider/FilterVarReturnTypeProvider.php:58]: I would actually either leave it here or add a default case to the switch below.

File: src/Psalm/Internal/Scanner/DocblockParser.php [hunk 2]
```diff
@@ -101,8 +101,10 @@ public static function parse(string $docblock, int $offsetStart): ParsedDocblock
                 unset($lines[$k]);
             } else {
                 // Strip the leading *, if present.
-                $lines[$k] = str_replace("\t", ' ', $line);
-                $lines[$k] = preg_replace('/^ *\*/', '', $line);
```
Comment (weirdan) [inline src/Psalm/Internal/Scanner/DocblockParser.php:105]: It looks like a bug actually, the tab expansion was futile.

**Review by weirdan — COMMENTED** [review]
> What I removed felt important but I couldn't get how it could be useful

We can leave it for another day then. I imagine it would be quite a pain to put it back half a year later should it actually break something.

File: src/Psalm/Internal/Provider/ReturnTypeProvider/FilterVarReturnTypeProvider.php (hunk 1 above)
Comment (orklah) [reply src/Psalm/Internal/Provider/ReturnTypeProvider/FilterVarReturnTypeProvider.php:58]: Did you see this line a little above? https://github.com/vimeo/psalm/pull/7233/files/0b3e1299decdd75fe6cf38acd88391cfea323516#diff-76c8a38ce19c782b153d6a5cbededfe163697de0e37f2f40559f3f07f8eaf318L50

It was just initialized twice

File: src/Psalm/Internal/Scanner/DocblockParser.php (hunk 2 above)
Comment (orklah) [reply src/Psalm/Internal/Scanner/DocblockParser.php:105]: Yeah, this was definitely a bug. Forgot to mention it in the PR. I fixed it the way I think it was intended, but I'm not sure what it was designed to do in what case though...

File: src/Psalm/Internal/Provider/ReturnTypeProvider/FilterVarReturnTypeProvider.php (hunk 1 above)
Comment (weirdan) [reply src/Psalm/Internal/Provider/ReturnTypeProvider/FilterVarReturnTypeProvider.php:58]: Ok then.

**Pull request merged**